NGitLab.Tests/SnippetsTest.cs

```diff
@@ -75,8 +75,45 @@ public async Task Test_snippet_inProject(VisibilityLevel visibility)
             snippetClient.Delete(newSnippet.ProjectId, returnedProjectSnippet.Id);
         }
 
+        [TestCase(VisibilityLevel.Private)]
+        [TestCase(VisibilityLevel.Internal)]
+        [TestCase(VisibilityLevel.Public)]
+        [NGitLabRetry]
+        public async Task Test_snippet_files(VisibilityLevel visibility)
+        {
+            using var context = await GitLabTestContext.CreateAsync();
+            var project = context.CreateProject();
+            var snippetClient = context.Client.Snippets;
+
+            var guid = Guid.NewGuid().ToString("N");
+            var snippetName = "testSnip" + guid;
+
+            // arrange
+            var newSnippet1 = new SnippetCreate
+            {
+                Title = snippetName,
+                Visibility = VisibilityLevel.Public,
+                Files = new[]
+                {
+                    new SnippetCreateFile() { Action = SnippetCreateFile.ActionType.Create, FilePath = "Path1.txt", Content = "Content1" },
+                    new SnippetCreateFile {  Action = SnippetCreateFile.ActionType.Create, FilePath = "Path2.txt", Content = "Content2" },
+                },
+            };
+
+            // act - assert
+            snippetClient.Create(newSnippet1);
+            var snippet = snippetClient.All.First();
```

Comment:
Tests can run in parallel, so you cannot be sure the snippet you get is the one you created in this test. It could be the one created by another test.

You can use the same logic as in the first test: https://github.com/ubisoft/NGitLab/pull/245/files/75f9baa1d6534cd2efab70bc70503267d0085bb0..bd08bfab21b604fb61bd9a3a86a74a6eb945dede#diff-2fe4fab06a5114d750bd61926c57591d0ed237feec7282522c740a10d28fa9bcR38